code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/lifecycle/LifecycleIntegrationTests.kt

```diff
@@ -0,0 +1,400 @@
+import com.adobe.marketing.mobile.services.HttpConnecting
+import com.adobe.marketing.mobile.services.Networking
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.ServiceProviderModifier
+import java.io.InputStream
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+
+private typealias NetworkMonitor = (url: String) -> Unit
+
+class LifecycleIntegrationTests {
+
+    companion object {
+        private var networkMonitor: NetworkMonitor? = null
+
+        @BeforeClass
+        @JvmStatic
+        fun setupClass() {
+            overrideNetworkService()
+        }
+
+        private fun overrideNetworkService() {
+            ServiceProvider.getInstance().networkService = Networking { request, callback ->
+                var connection: HttpConnecting = MockedHttpConnecting(null)
+                with(request.url) {
+                    when {
+                        startsWith("https://adobe.com") && contains("rules_lifecycle.zip") -> {
+                            connection = MockedHttpConnecting("rules_lifecycle")
+                        }
+                    }
+                }
+                if (callback != null) {
+                    callback.call(connection)
+                } else {
+                    // If no callback is passed by the client, close the connection.
+                    connection.close()
+                }
+                networkMonitor?.let { it(request.url) }
+            }
+
+        }
+    }
+
+    @Before
+    fun setup() {
+        networkMonitor = null
+        SDKHelper.resetSDK()
+        clearSharedPreference()
+
+        val countDownLatch = CountDownLatch(1)
+        MobileCore.setApplication(ApplicationProvider.getApplicationContext())
+        MobileCore.setLogLevel(LoggingMode.VERBOSE)
+        MobileCore.registerExtensions(
+            listOf(
+                Lifecycle.EXTENSION,
+                Identity.EXTENSION,
+                Signal.EXTENSION,
+                MonitorExtension::class.java
+            )
+        )
+        {
+            countDownLatch.countDown()
+        }
+        assertTrue(countDownLatch.await(1000, TimeUnit.MILLISECONDS))
+    }
+
+    private fun clearSharedPreference() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        val sharedPreference = context.getSharedPreferences("AdobeMobile_Lifecycle", 0)
+        val editor = sharedPreference.edit()
+        editor.clear()
+        editor.commit()
+    }
+
+    @Test
+    fun testInstall() {
+        // setup
+        val countDownLatch = CountDownLatch(1)
+        networkMonitor = { url ->
+            if (url.startsWith("https://www.lifecycle.com")) {
+                // verify
+                assertTrue(url.contains("installevent=InstallEvent"))
```

Comment:
Can you verify if assert failure actually fails the test as this happens in background thread?

Reply:
The `latch.countdown` on the next line does not run if the assert fails so the test never terminates. Added a timeout to the test.

Reply:
Can you capture the variable and assert it outside the block? That way we will easily know about the failure.